src/msg/async/frames_v2.h

```diff
@@ -561,20 +603,70 @@ struct MessageHeaderFrame
     return this->payload;
   }
 
-  static MessageHeaderFrame Encode(const ceph_msg_header2 &msg_header,
-                                   const uint32_t front_len,
-                                   const uint32_t middle_len,
-                                   const uint32_t data_len)  {
+  static MessageHeaderFrame Encode(ceph::crypto::onwire::rxtx_t &session_stream_handlers,
```

Comment:
Looks like this is no longer just a `MessageHeaderFrame`. Rename this to `MessageFrame` ?

Reply:
Yeah, that's true. I think that introducing segments diverged the `Frame` hierarchy/model far away from what we have on wire. I feel that `SignedEncryptedFrame` doesn't fit anymore (it's only used for control, single-segment frames; `Message{Header}Frame` bypasses it entirely). The same with single `payload` field of `Frame`. What we have now is closer to:

```cpp
template <class T, __u8 SegmentsNumV>
struct Frame {
protected:
  std::array<ceph::bufferlist, SegmentsNumV> segments;
  static_assert(std::size(segments) <= MAX_NUM_SEGMENTS);
```

Will fix the naming.